Assets/Scripts/Maze/Generation/GenerationMediator.cs

```diff
@@ -18,19 +18,23 @@ public class GenerationMediator : MonoBehaviour
 
         private void OnEnable()
         {
-            _button.onClick.AddListener(GenerateMaze);
+            _button.onClick.AddListener(() => GenerateMaze());
             MazeGenerator.OnGenerated += DisplaySeed;
         }
 
         private void OnDisable()
         {
-            _button.onClick.RemoveListener(GenerateMaze);
+            _button.onClick.RemoveAllListeners();
```

Comment:
I wouldn't do this because this could cause a lot of issues, and delegate listeners cannot be removed anyway.

Reply:
Removed it